byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +71,110 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
-function addToStops(stop){
-  console.log("add" + stop.placename + "to the ArrayList in the servlet");
+/** Clear the stops panel in the html */
+function clearStops(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+    stopList.innerText = "";
+  }
 }
 
-function getRecs() {
-  fetch('/api/recs')
+/** Get the new list of stops from datastore onload */
+function getStopsOnload(){
+  clearStops();
+  fetch('/api/stop')
   .then(response => response.json())
-  .then((recs) => {
-    const recList = document.getElementById('rec-list');
-    recList.innerText = ""; // clear old comments 
-    recs.forEach((stop) => {
-      let btn = document.createElement("button");
-      btn.innerText = stop.placename;
-      btn.setAttribute("class", "btn rec-btn");
-      btn.addEventListener("click", function() {
-        addToStops(stop);
-      });
-      recList.appendChild(btn);
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      stops.push(stop);
     });
+    renderStopsList();
+  });
+}
+
+/** Re-render stop list synchronously */
+function renderStopsList(){
+  clearStops();
+  stops.forEach((stop)=>{
+    renderStop(stop);
+  })
+}
+
+/** Render stop list  
+ *  @param {String} stop a String to add as a button in the schedule panel in the html
+ */
+function renderStop(stop){
```

Comment:
Same advice as below for using a `createStopButton` and letting the caller decide where it should go on the page.

Reply:
Done!